### ISSUE TITLE
Modal is not accessible

### ISSUE THREAD

**Issue by adamwoodnz** [uikit/uikit#1420]
While the modal manages aria state it does not manage focus, meaning it is inaccessible for keyboard navigation.

When opening a modal it needs to be focused so that the user can tab through it's content. When the user reaches the end of the modal content, the focus should not be able to move back into the parent page, instead it should be cycled back to the start of the modal.

On closing the modal the element which opened the modal should be focused again, so that the user can continue where they left off.

There is a good summary article available [here](http://www.smashingmagazine.com/2014/09/making-modal-windows-better-for-everyone/), and the W3C draft guideline [here](http://www.w3.org/WAI/PF/aria-practices/#dialog_modal).

**Comment by chiefie** [comment]
+1

**Comment by samueljseay** [comment]
+1 for this. Overall I think uikit accessibility is pretty good. this would really help.

**Comment by luetn** [comment]
+1

**Comment by samueljseay** [comment]
Here's a suggested fix in jsfiddle: http://jsfiddle.net/samueljseay/Las4z36u/49/

This remembers last element focused before modal opened and returns to it on close. It also makes tab indexing circular within the modal.

It includes borrowing the ":tabbable" selector from jquery ui (a very small additional dependency) and some tweaks to the hide and show methods.

To the maintainers: would you consider a PR with these fixes? I'm a big fan of uikit and I'd love to contribute. :smile_cat:

**Comment by dteirney** [comment]
+1

**Comment by phazor** [comment]
+1. Any updates on this?

**Comment by samueljseay** [comment]
@saschadube I don't mean to be a pain, but I really like UIKit but get the general feeling that the team is ignoring a lot of the incoming suggestions and needed fixes from developers. Is this gonna get some attention or just be ignored?

**Comment by leik** [comment]
+1

**Comment by philoest** [comment]
The modal component has been reworked in UIkit3, please check, if these issues still persist in version 3 and if so, please open a new issue.
